### DISCUSSION TITLE
How can I use GPU on training in Linux.

### DISCUSSION THREAD

**Discussion by youngjae142** [mir-group/nequip#181] [Q&A] [answered]
Hello!
I successfully installed nappy as instructed, and two tests were all successful.

Two tests:
nequip-train configs/minimal.yaml
pytest tests/unit/
(pytest tests/unit --> uses GPU and no error occurred)

But "nequip-train configs/minimal.yaml" test always says:
Torch device: cpu
not cuda.
Our computer environment: CentOS7, intel CPU, NVIDIA GPU (K80, V100, RTX3090, tested all of them.)
Also set the CUDA_VISIBLE_DEVICES=0

Is there any parameter or environment variable I have to set?

**Comment by simonbatzner** [comment, marked as answer]
Hi @ssrokyz, there is an explicit setting: 

```
device:  cuda   
```

But the default is true and minimal.yaml doesn't set it explicitly, hence it defaults to cuda. If you ssh onto your GPU and run 

```
torch.cuda.is_available()
```

what does it say?

**Reply by youngjae142** [reply]
It says "False"
So it means PyTorch is not properly installed to use GPU, maybe?
I'll try to reinstall it!
I installed PyTorch with Conda since I need to install cudatoolkit with >=11.0.
I have 3090, and cudatoolkit with >= 11.0 is needed.

**Reply by Linux-cpp-lisp** [reply]
Yes, this means that PyTorch is not installed to use GPU.

**Comment by Linux-cpp-lisp** [comment]
> (pytest tests/unit --> uses GPU and no error occurred)

How do you know it used GPU?

**Reply by youngjae142** [reply]
By nvidia-smi command that shows the status of GPUs, and one GPU process stays occupied for the overall test process.
Maybe, cudatoolkit setup is successful, but PyTorch does not use it.

**Comment by youngjae142** [comment]
> Hi @ssrokyz, there is an explicit setting:
> 
> ```
> device:  cuda   
> ```
> 
> But the default is true and minimal.yaml doesn't set it explicitly, hence it defaults to cuda. If you ssh onto your GPU and run
> 
> ```
> torch.cuda.is_available()
> ```
> 
> what does it say?

The problem was the PyTorch installation.
Installation from scratch (First conda install cudatoolkit >= 11.0, then, pip install in NequIP source folder) solved the problem.

Thank you very much @simonbatzner and @Linux-cpp-lisp  for the instant and good answer!

**Reply by simonbatzner** [reply]
Glad it's fixed.
